src/Mvc/Mvc.RazorPages/src/Infrastructure/PageLoaderBase.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    /// <summary>
+    /// Creates a <see cref="CompiledPageActionDescriptor"/> from a <see cref="PageActionDescriptor"/>.
+    /// </summary>
+#pragma warning disable CS0618 // Type or member is obsolete
+    public abstract class PageLoaderBase : IPageLoader
+#pragma warning restore CS0618 // Type or member is obsolete
+    {
+        public abstract ValueTask<CompiledPageActionDescriptor> LoadAsync(PageActionDescriptor actionDescriptor);
+
+        CompiledPageActionDescriptor IPageLoader.Load(PageActionDescriptor actionDescriptor)
```

Comment:
Does this need to be obsolete too? or doesn't matter.

Reply:
It's private isn't it? (as it is an explicit interface implementation)